specification/communication/data-plane/CallingServer/preview/2022-04-07-preview/communicationservicescallingserver.json

```diff
@@ -1056,6 +1153,179 @@
+          "type": "integer"
+        },
+        "maxTonesToCollect": {
+          "format": "int32",
+          "description": "Maximum number of DTMFs to be collected.",
+          "type": "integer"
+        },
+        "stopTones": {
+          "description": "List of tones that will stop recognizing.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CommunicationIdentifier": {
+      "description": "Identifies a participant in Azure Communication services. A participant is, for example, a phone number or an Azure communication user.",
+      "type": "object",
+      "properties": {
+        "rawId": {
+          "description": "The full Id of the identifier.",
+          "type": "string"
+        },
+        "communicationUser": {
+          "$ref": "#/definitions/CommunicationUserIdentifier",
+          "description": "The communication user."
+        },
+        "phoneNumber": {
+          "$ref": "#/definitions/PhoneNumberIdentifier",
+          "description": "The phone number."
+        },
+        "microsoftTeamsUser": {
+          "$ref": "#/definitions/MicrosoftTeamsUserIdentifier",
+          "description": "The Microsoft Teams user."
+        }
+      }
+    },
+    "CommunicationUserIdentifier": {
+      "description": "A user that got created with an Azure Communication Services resource.",
+      "required": [
+        "id"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The Id of the communication user.",
+          "type": "string"
+        }
+      }
+    },
+    "PhoneNumberIdentifier": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The phone number in E.164 format.",
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationCloudEnvironment": {
+      "description": "The cloud that the Microsoft Teams user belongs to. By default 'public' if missing.",
+      "enum": [
+        "public",
+        "dod",
+        "gcch"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CommunicationCloudEnvironment",
+        "modelAsString": true
+      }
+    },
+    "MicrosoftTeamsUserIdentifier": {
+      "description": "A Microsoft Teams user.",
+      "required": [
+        "userId"
+      ],
+      "type": "object",
+      "properties": {
+        "userId": {
+          "description": "The Id of the Microsoft Teams user. If not anonymous, this is the AAD object Id of the user.",
+          "type": "string"
+        },
+        "isAnonymous": {
+          "description": "True if the Microsoft Teams user is anonymous. By default false if missing.",
+          "type": "boolean"
+        },
+        "cloud": {
+          "$ref": "#/definitions/CommunicationCloudEnvironment"
+        }
+      }
+    },
```

Comment:
The identifiers models can be referenced from the [common 2022-07-13](https://github.com/Azure/azure-rest-api-specs/blob/0a4e1e98112a37f70e46a26ec1598d41d8c4b363/specification/communication/data-plane/Common/stable/2022-07-13/common.json#L87-L160), including the `CommunicationCloudEnvironment`.